docs/sources/source/index.rst

```diff
@@ -1,5 +1,5 @@
-NVIDIA Neural Modules: NeMo
-===========================
+NVIDIA Neural Modules Developer Guide
```

Comment:
I'd try to keep NeMo in there, e.g. `NVIDIA NeMo (Neural Modules) Developer Guide`